src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,62 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  selectedFilter: string;
+  onFilterSelect: (selectedFilter: string) => void;
+  query: string;
+  onQueryChange: (query: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = React.memo(
+  ({
+    selectedFilter,
+    onFilterSelect,
+    query,
+    onQueryChange,
+  }) => (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectedFilter}
+            onChange={(event) => onFilterSelect(event.currentTarget.value)}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={(event) => onQueryChange(event.currentTarget.value)}
         />
-      </span>
-    </p>
-  </form>
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => {
+                onQueryChange('');
+                onFilterSelect('all');
```

Comment:
use enum and rename methods
They should show what actions will be done

`onClick` -> `selectFilterType(....)`

but `onClick` -> `onFilterSelect(...)` sounds wierd
```suggestion
                selectFilterType(FilterType.ALL);
```

Reply:
Done!